# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsBaseWidget.cpp
# Commit: 786823305560
# Full Hash: 786823305560b03316cb758207f76838388603af
# Author: sotaro <sotaro.ikeda.g@gmail.com>
# Date: 2020-05-05 03:49:49
# Regressor Bug: 1574746
# File Overlap Count: 1
# Description:
#   Bug 1574746 - Remove AllowWebRenderForThisWindow() r=nical
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D71440
# ==============================================================================

diff -r 7af7b4435997 -r 786823305560 widget/nsBaseWidget.cpp
--- a/widget/nsBaseWidget.cpp	Mon May 04 23:13:55 2020 +0000
+++ b/widget/nsBaseWidget.cpp	Sun May 03 11:38:53 2020 +0000
@@ -823,12 +823,6 @@
 }
 
 bool nsBaseWidget::ComputeShouldAccelerate() {
-  if (gfx::gfxVars::UseWebRender() && !AllowWebRenderForThisWindow()) {
-    // If WebRender is enabled, non-WebRender widgets use the basic compositor
-    // (at least for now), even though they would get an accelerated compositor
-    // if WebRender wasn't enabled.
-    return false;
-  }
   return gfx::gfxConfig::IsEnabled(gfx::Feature::HW_COMPOSITING) &&
          WidgetTypeSupportsAcceleration();
 }
@@ -841,13 +835,6 @@
             StaticPrefs::apz_popups_enabled())));
 }
 
-bool nsBaseWidget::AllowWebRenderForThisWindow() {
-  return WindowType() == eWindowType_toplevel ||
-         WindowType() == eWindowType_child ||
-         WindowType() == eWindowType_dialog ||
-         (WindowType() == eWindowType_popup && HasRemoteContent());
-}
-
 void nsBaseWidget::CreateCompositor() {
   LayoutDeviceIntRect rect = GetBounds();
   CreateCompositor(rect.Width(), rect.Height());
@@ -1213,9 +1200,8 @@
     // If widget type does not supports acceleration, we use ClientLayerManager
     // even when gfxVars::UseWebRender() is true. WebRender could coexist only
     // with BasicCompositor.
-    bool enableWR = gfx::gfxVars::UseWebRender() &&
-                    WidgetTypeSupportsAcceleration() &&
-                    AllowWebRenderForThisWindow();
+    bool enableWR =
+        gfx::gfxVars::UseWebRender() && WidgetTypeSupportsAcceleration();
     bool enableAPZ = UseAPZ();
     CompositorOptions options(enableAPZ, enableWR);
 
